src/main/java/com/parasoft/parabank/web/controller/RestServiceProxyController.java

```diff
@@ -312,18 +317,58 @@ private void authenticate() throws Exception {
             return;
         }
         ServletRequestAttributes attr = (ServletRequestAttributes) RequestContextHolder.currentRequestAttributes();
-        HttpSession session = attr.getRequest().getSession(false);
+        HttpServletRequest request = attr.getRequest();
+        String authHeader = request.getHeader("Authorization");
         Locale locale = LocaleContextHolder.getLocale();
-        if (session == null) {
-            throw new AuthenticationException(messageSource.getMessage("user.login.required", null, locale));
-        }
-        Object result = session.getAttribute("userSession");
-        if (result == null || !(result instanceof UserSession)) {
-            throw new AuthenticationException(messageSource.getMessage("user.login.required", null, locale));
-        }
-        UserSession userSession = (UserSession) result;
-        if (userSession.getCustomer() == null) {
-            throw new AuthenticationException(messageSource.getMessage("user.login.required", null, locale));
+        if(authHeader == null) {
+        	HttpSession session = request.getSession(false);
+            if (session == null) {
+                throw new AuthenticationException(messageSource.getMessage("user.login.required", null, locale));
+            }
+            Object result = session.getAttribute("userSession");
+            if (result == null || !(result instanceof UserSession)) {
+                throw new AuthenticationException(messageSource.getMessage("user.login.required", null, locale));
+            }
+            UserSession userSession = (UserSession) result;
+            if (userSession.getCustomer() == null) {
+                throw new AuthenticationException(messageSource.getMessage("user.login.required", null, locale));
+            }
+        } else {
+        	StringTokenizer st = new StringTokenizer(authHeader);
+			if (st.hasMoreTokens()) {
+				String basic = st.nextToken();
+				if (basic.equalsIgnoreCase("Basic")) {
+					try {
+						String credentials = new String(Base64.decode(st.nextToken()), "UTF-8");
+						int p = credentials.indexOf(":");
+						if (p != -1) {
+							String username = credentials.substring(0, p).trim();
+							String password = credentials.substring(p + 1).trim();
+
+							if (username == null || username.length() <= 0 || password == null
```

Comment:
You don't need to check username == null because String.trim() will never return null.

Reply:
I am just re-using the check was in LoginController as part of authentication. I will make the change to both places.